SingularityExecutor/src/main/java/com/hubspot/singularity/executor/task/UnixLocalDownloadServiceFetcher.java

```diff
@@ -0,0 +1,153 @@
+package com.hubspot.singularity.executor.task;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Lists;
+import com.hubspot.deploy.S3Artifact;
+import com.hubspot.mesos.JavaUtils;
+import com.hubspot.singularity.executor.config.SingularityExecutorConfiguration;
+import com.hubspot.singularity.s3.base.ArtifactDownloadRequest;
+import com.hubspot.singularity.s3.base.config.SingularityS3Configuration;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.util.BytesContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+
+public class UnixLocalDownloadServiceFetcher implements LocalDownloadServiceFetcher {
+  private final ObjectMapper objectMapper;
+  private final HttpClient httpClient;
+  private final SingularityExecutorConfiguration executorConfiguration;
+  private final SingularityS3Configuration s3Configuration;
+  private final String localDownloadPath;
+
+  public UnixLocalDownloadServiceFetcher(
+    HttpClient httpClient,
+    ObjectMapper objectMapper,
+    SingularityExecutorConfiguration executorConfiguration,
+    SingularityS3Configuration s3Configuration
+  ) {
+    this.httpClient = httpClient;
+    this.objectMapper = objectMapper;
+    this.executorConfiguration = executorConfiguration;
+    this.s3Configuration = s3Configuration;
+    this.localDownloadPath =
+      String.format("http://localhost%s", s3Configuration.getLocalDownloadPath());
+  }
+
+  @Override
+  public void start() {
+    try {
+      httpClient.start();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void stop() {
+    try {
+      httpClient.stop();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public String getDownloadPath() {
+    return localDownloadPath;
+  }
+
+  @Override
+  public void downloadFiles(
+    List<? extends S3Artifact> s3Artifacts,
+    SingularityExecutorTask task
+  )
+    throws InterruptedException {
+    final List<CompletableFutureHolder> futures = Lists.newArrayListWithCapacity(
```

Comment:
Could possibly refactor the two local download fetchers into a single implementation that takes in an S3Artifact -> CompletableFuture function, if you think consolidating the implementations is worth it.

Reply:
The ning async client works with the other futures by default. Didn't want to mess with the existing/working code as much as possible for the moment. Agree CompletableFuture is much nicer though